src/components/EditCoursePage/index.jsx

```diff
@@ -0,0 +1,213 @@
+/* eslint-disable camelcase */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Helmet from 'react-helmet';
+
+import EditCourseForm from './EditCourseForm';
+import StatusAlert from '../StatusAlert';
+import LoadingSpinner from '../LoadingSpinner';
+
+
+class EditCoursePage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      startedFetching: false,
+    };
+    this.handleCourseEdit = this.handleCourseEdit.bind(this);
+    this.setStartedFetching = this.setStartedFetching.bind(this);
+  }
+
+  componentDidMount() {
+    this.props.fetchCourseInfo();
+    this.props.fetchCourseOptions();
+    this.setStartedFetching();
+  }
+
+  setStartedFetching() {
+    this.setState({ startedFetching: true });
+  }
+
+  handleCourseEdit(options) {
+    /*
+      Need to do some preprocessing before sending anything to course-discovery.
+      This includes:
+        1. Only including subjects that have values
+        2. Putting the entitlement into an array and also adding in the sku
+          (required for updating price)
+        3. Renaming the image and video fields to correspond to what course-discovery is expecting
+        4. Setting the uuid so we can create the url to send to course-discovery
+    */
+    const {
+      courseInfo: {
+        data: {
+          uuid,
+          entitlements,
+        },
+      },
+    } = this.props;
+    const courseData = options;
+    courseData.subjects = [
+      courseData.subjectPrimary,
+      courseData.subjectSecondary,
+      courseData.subjectTertiary,
+    ].filter(subject => !!subject);
+    courseData.entitlements = entitlements && [{
+      mode: courseData.mode,
+      price: courseData.price,
+      sku: entitlements[0].sku,
+    }];
+    courseData.image = courseData.imageSrc;
+    courseData.video = courseData.videoSrc;
+    courseData.uuid = uuid;
+    return this.props.editCourse(courseData);
+  }
+
+  render() {
+    if (!this.props.courseInfo || !this.props.courseOptions) {
+      return (
+        <StatusAlert
+          id="error"
+          alertType="danger"
+          title="Course Edit Form failed to load: "
+          message="Course information unavailable."
+        />
+      );
+    }
+
+    const {
+      courseInfo,
+      courseInfo: {
+        data: {
+          title,
+          key,
+          short_description,
+          full_description,
+          outcome,
+          subjects,
+          image,
+          prerequisites_raw,
+          level_type,
+          learner_testimonials,
+          faq,
+          additional_information,
+          syllabus_raw,
+          video,
+          entitlements,
+        },
+      },
+      courseOptions,
+    } = this.props;
+
+    // If we want to keep a lot of the logic in the lower return,
+    // we have to do all this weird '&&' logic to make sure the data is there.
```

Comment:
We might want to avoid calling the logic "weird" in the comment, makes it seems like we should remove it.